File: Data_Structures/Binary_Tree/Implementation.py

class Node:
    def __init__(self, value):
        self.left = None
        self.right = None
        self.value = value

class BinaryTree:
    def __init__(self, value):
        self.root = Node(value)

    def addRecursive(self, root, value):
        if root == None:
            return Node(value)
        if root.value < value:
            root.right = self.addRecursive(root.right, value)
        elif root.value > value:
            root.left = self.addRecursive(root.left, value)
        return root

    def add(self, value):
        self.root = self.addRecursive(self.root, value)



if __name__ == '__main__':
    bt = BinaryTree(4)
    bt.add(5)
    bt.add(6)
    bt.add(10)
    bt.add(3)

    print(bt.root.left.value)

File: Data_Structures/Binary_Tree/implementation.java
class Node{
    int value;
    Node left;
    Node right;

    Node(int value) {
        this.value = value;
        right = null;
        left = null;
    }
}

class BinaryTree{
  Node root;

  public BinaryTree(){
    root = null;
  }

  public BinaryTree(int value){
    root = new Node(value);
  }

  public void add(int value) {
    root = addRecursive(root, value);
}

  public Node addRecursive(Node node, int value){
      if (node == null){
        return new Node(value);
      }
      if(node.value < value){
          node.right = addRecursive(node.right, value);
      }else if(node.value > value){
        node.left = addRecursive(node.left, value);
      }else{
        return node;
      }
      return node;
  }
}

public class Implementation{
    public static void main(String [] args){
      BinaryTree bt = new BinaryTree();
      bt.add(6);
      bt.add(4);
      bt.add(8);
      bt.add(3);
      bt.add(5);
      bt.add(7);
      bt.add(9);
      NodeA j = new NodeA(1);

      System.out.println(bt.root.left.right.value);
    }

}
